core/src/main/java/org/pentaho/di/core/Const.java

```diff
@@ -1717,10 +1717,14 @@ public static long round( long f, int places, int roundingMode ) {
    */
   public static int toInt( String str, int def ) {
     int retval;
-    try {
-      retval = Integer.parseInt( str );
-    } catch ( Exception e ) {
+    if ( str == null ) {
       retval = def;
```

Comment:
is this just copying Apache's common lang NumberUtils function ?

 https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/math/NumberUtils.html#toInt-java.lang.String-int-